app/src/main/java/com/communistutopia/spacetrader/model/Universe.kt

```diff
@@ -0,0 +1,78 @@
+package com.communistutopia.spacetrader.model
+
+import kotlin.random.Random
+import kotlin.math.pow
+import kotlin.math.sqrt
+
+/**
+ * Singleton object that handles all the logic for generating universe and other events
+ */
+object Universe {
+
+    // Constants
+    private val X_SIZE: Int = 150
+    private val Y_SIZE: Int = 100
+    private val NUM_SYSTEMS: Int = 10
+
+    private val solarSystems: ArrayList<SolarSystem> = ArrayList()
+    private val universe: HashMap<Pair<Int, Int>, String> // Initializing a 2D array is the most frustrating ordeal. So this is a hacky solution
+    private val random: Random
+
+
+    init {
+        universe = HashMap()
+        random = Random.Default
+    }
+
+    fun generateUniverse() {
+        var i: Int = NUM_SYSTEMS
+        while (i > 0) {
+            var x = random.nextInt(X_SIZE)
+            var y = random.nextInt(Y_SIZE)
+            val pair = Pair(x, y)
+            if (!universe.containsKey(pair)) {
+                var name: String = "System%d".format(i)
+                solarSystems.add(SolarSystem(planet = Planet(""), name = name, x = x, y = y))
+                universe[pair] = name
+            }
+        }
+    }
+}
+
+class SolarSystem(planet: Planet, name: String, tech: Int = 0, x: Int, y: Int) {
+
+    var planet: Planet
```

Comment:
We also talked about having multiple planets per SS. Perhaps this could be refactored as a Collection of planets? Unsure about how that would affect your randomization logic